Scripts/build/Jenkins/Jenkinsfile

```diff
@@ -0,0 +1,665 @@
+    // instances. This allow us to scale up and down without having to re-download 3rdParty
+    envVarMap['LY_PACKAGE_DOWNLOAD_CACHE_LOCATION'] = "${envVarMap['WORKSPACE']}/3rdParty/downloaded_packages"
+    envVarMap['LY_PACKAGE_UNPACK_LOCATION'] = "${envVarMap['WORKSPACE']}/3rdParty/packages"
+
+    return envVarMap
+}
+
+def GetEnvStringList(Map envVarMap) {
+    def strList = []
+    envVarMap.each { var ->
+        strList.add("${var.key}=${var.value}")
+    }
+    return strList
+}
+
+def getEngineRemoteConfig(remoteConfigs) {
+    def engineRemoteConfigs = [name: "${ENGINE_REPOSITORY_NAME}",
+        url: remoteConfigs.url[0]
+            .replace("${PROJECT_REPOSITORY_NAME}", "${ENGINE_REPOSITORY_NAME}")
+            .replace("/${PROJECT_ORGANIZATION_NAME}/", "/${ENGINE_ORGANIZATION_NAME}/"),
+        credentialsId: remoteConfigs.credentialsId[0]
+        ]
+    return engineRemoteConfigs
+}
+
+def CheckoutBootstrapScripts(String branchName) {
+    checkout([$class: 'GitSCM',
+        branches: [[name: "*/${branchName}"]],
+        doGenerateSubmoduleConfigurations: false,
+        extensions: [
+            [$class: 'PruneStaleBranch'],
+            [$class: 'AuthorInChangelog'],
+            [$class: 'SparseCheckoutPaths', sparseCheckoutPaths: [
+                [ $class: 'SparseCheckoutPath', path: 'scripts/build/Jenkins/' ],
+                [ $class: 'SparseCheckoutPath', path: 'scripts/build/bootstrap/' ],
+                [ $class: 'SparseCheckoutPath', path: 'scripts/build/Platform' ]
+            ]],
+            // Shallow checkouts break changelog computation. Do not enable.
+            [$class: 'CloneOption', noTags: false, reference: '', shallow: false]
+        ],
+        submoduleCfg: [],
+        userRemoteConfigs: [getEngineRemoteConfig(scm.userRemoteConfigs)]
+    ])
+}
+
+def CheckoutRepo(boolean disableSubmodules = false) {
+
+    def projectsAndUrl = [
+        "${ENGINE_REPOSITORY_NAME}": getEngineRemoteConfig(scm.userRemoteConfigs),
+        "${PROJECT_REPOSITORY_NAME}": scm.userRemoteConfigs[0]
+    ]
+
+    projectsAndUrl.each { projectAndUrl ->
+        if(!fileExists(projectAndUrl.key)) {
+            palMkdir(projectAndUrl.key)
+        }
+        dir(projectAndUrl.key) {           
+            if(fileExists('.git')) {
+                // If the repository after checkout is locked, likely we took a snapshot while git was running,
+                // to leave the repo in a usable state, garbage collect.
+                def indexLockFile = '.git/index.lock'
+                if(fileExists(indexLockFile)) {
+                    palSh('git gc', 'Git GarbageCollect')    
+                }
+                if(fileExists(indexLockFile)) { // if it is still there, remove it
+                    palRm(indexLockFile)
+                }
+            }
+        }
+    }
+
+    def random = new Random()
+    def retryAttempt = 0
+    retry(5) {

```

Comment:
nit: Magic number, no comment as to why/how

Reply:
Yeah, I'm seeing this magic number in all the existing Jenkins config files (engine and the atom project repo). Not quite sure why we want to retry 5 times and might be a request for the infra to team to add comment for this magic number.

Reply:
As I understand it, this is a arbitrary retry number. We can move it to a named variable in another PR (most likely through a shared library)